api/v1beta2/helmrepository_types.go

```diff
@@ -80,6 +80,11 @@ type HelmRepositorySpec struct {
 	// +optional
 	Timeout *metav1.Duration `json:"timeout,omitempty"`
 
+	// InsecureSkipTLSVerify skips the validation of the TLS certificate of the
+	// OCI registry endpoint.
+	// +optional
+	InsecureSkipTLSVerify bool `json:"insecureSkipTLSverify,omitempty"`
```

Comment:
```suggestion
	Insecure bool `json:"insecure,omitempty"`
```

As specified in RFC-0004 https://github.com/fluxcd/flux2/tree/main/rfcs/0004-insecure-http#objects

Reply:
the thing is we do not set `plain-http` here. We just disable tls verification on the client side.

Reply:
But Flux doesn't support this, we decided that people must give Flux the custom CA. Not sure why this PR then.

Reply:
I was not aware of that. This PR is still relevant as it enables users to provide custom certs.

Reply:
Ok so let's drop this field from the CRD and keep the CA code only? cc @hiddeco

Reply:
This sounds good to me :heavy_plus_sign: